planex/cmd/fetch.py

```diff
@@ -115,35 +103,42 @@ def fetch_http(url, filename, retries):
     Download the file at url and store it as filename
     """
 
-    while True:
-        retries -= 1
-        try:
-            url_string = urlparse.urlunparse(url)
-            logging.debug("Fetching %s to %s", url_string, filename)
+    url_string = urlparse.urlunparse(url)
+    logging.debug("Fetching %s to %s", url_string, filename)
+
+    useragent = "planex-fetch/%s" % pkg_resources.require("planex")[0].version
+
+    # We need this because centos ships requests 2.8.0
+    # It can be greatly simplified with requests >= 2.18.0
+    headers = requests.utils.default_headers()
+    headers.update({
+        "user-agent": useragent,
+    })
+
+    # Once we use requests >= 2.18.0, we should change this into
+    # with requests.get ... as r:
+    req = requests_retry_session(retries).get(
+        url_string, headers=headers, timeout=30, stream=True)
+    req.raise_for_status()
 
-            tmp_filename = filename + "~"
-            with open(tmp_filename, "wb") as tmp_file:
-                curl_get(url_string, tmp_file)
-                best_effort_file_verify(tmp_filename)
-                shutil.move(tmp_filename, filename)
-                # Write an origin file for tracking.
-                with open('{0}.origin'.format(filename), 'w') as origin_file:
-                    origin_file.write('{0}\n'.format(url_string))
-                return
+    tmp_filename = filename + "~"
+    with open(tmp_filename, 'wb') as out:
+        shutil.copyfileobj(req.raw, out)
+        best_effort_file_verify(tmp_filename)
```

Comment:
Note that this is not 100% equivalent to the old code, however this best effort verification will happen only if we were able to complete the download without consuming all the retries, in that case we should have received the complete file, otherwise we would have failed before